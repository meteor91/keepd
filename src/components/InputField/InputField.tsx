import React, { useId } from 'react';
import clsx from 'clsx';
import { Input, type InputProps } from '../Input';
import styles from './InputField.module.less';

export interface InputFieldProps extends InputProps {
    label?: string;
    message?: string;
    name?: string;
}

export const InputField: React.FC<InputFieldProps> = (props) => {
    const {
        label,
        message,
        error,
        name,
        ...rest
    } = props;

    const id = useId();
    const inputId = `${name ?? 'input'}-${id}`;

    return (
        <div className={styles.formInput}>
            {label && (
                <label className={styles.label} htmlFor={inputId}>
                    {label}
                </label>
            )}

            <Input
                error={error}
                id={inputId}
                {...rest}
            />

            {message && (
                <div className={clsx(styles.message, error && styles.error)} role="alert">
                    {message}
                </div>
            )}
        </div>
    );
};
