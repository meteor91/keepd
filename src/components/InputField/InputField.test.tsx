import React from 'react';
import { render } from '@testing-library/react';
import { InputField } from './InputField';

describe('InputField', () => {
    it('should render correct', () => {
        const { container } = render(
            <InputField
                value="test"
                label="test field"
                message="test message"
            />,
        );

        expect(container.firstChild).toMatchSnapshot();
    });
});