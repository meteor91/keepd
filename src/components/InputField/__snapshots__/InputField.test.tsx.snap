// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField should render correct 1`] = `
<div
  class="formInput"
>
  <label
    class="label"
    for="input-:r0:"
  >
    test field
  </label>
  <input
    class="input"
    id="input-:r0:"
    readonly=""
    value="test"
  />
  <div
    class="message"
    role="alert"
  >
    test message
  </div>
</div>
`;
